### PR TITLE
Finalise and open report command

### LINKED ISSUE #48: [MUST] As an interviewer, I want to be able to finalise the attributes and questions for an interview session so that I do not change the rubrics for the interview session between interviews
After which the questions and attributes could not be changed

### LINKED ISSUE #70: [MUST] As an interviewer, I would like to be able to open the report of an interview with a particular interviewee that I have interviewed so that I can refer back to the result of the interview

### LINKED ISSUE #71: Retrieving the result of an interview
As an interviewer, after interviewing an interviewee, I would like to be able to see the result of the interview of a particular interviewee. Through the result of the interview, I would like to be able to

1. see all the remarks and the time it was created during the interview. 
2. navigate to the answer of a particular question (answers are recorded as remarks)
3. navigate to a particular time of the interview session

#70 
#69 
#68

### PR DESCRIPTION
**Pull request by agnesnatasya**
Implemented Finalise command for initialisation phase such that the interviewee cannot further change the attributes, questions and interviewees of an interview session. This creates some behavioural changes in the other command classes  of the CRUD operations as well.

Implemented OpenReport command for opening report of an interview after an interview session.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/OpenReportCommand.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        IntervieweeList interviewees = model.getIntervieweeList();
+        try {
+            Interviewee identifiedInterviewee = interviewees.getInterviewee(identifier);
+            identifiedInterviewee
+                    .getTranscript()
+                    .orElseThrow(() -> new IllegalActionException(
+                            String.format("Interviewee %1$s has not been interviewed.", identifier)));
```
Comment (CornCobs) [inline src/main/java/seedu/address/logic/commands/OpenReportCommand.java]: Throw CommandException here instead. Then you don't even need to have a try-catch block. IllegalActionException should only be thrown inside Model, not in Commands

File: src/main/java/seedu/address/logic/commands/OpenReportCommand.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * OpenReportCommand describes the behavior when the
+ * client wants to open an interview report of an interviewee.
+ */
+public class OpenReportCommand extends Command {
+    public static final String COMMAND_WORD = "report";
+    public static final String MESSAGE_SUCCESS = "Interview report opened: %1$s";
+    public static final String MESSAGE_USAGE = "open " + COMMAND_WORD
+            + ": Opens the interview report of an interviewee."
+            + "Example: list " + COMMAND_WORD + " John Doe";
+
```
Comment (CornCobs) [inline src/main/java/seedu/address/logic/commands/OpenReportCommand.java]: Uh, so is the command word "report", "list" or "open"? According to the Parser, it is `open <identifier>`, but here it seems to be `open report`, but in the Example on line 20, it says `list report John Doe`

**Review by CornCobs — COMMENTED** [review]
I think it is best to directly throw CommandException in the Commands instead of IllegalActionException.
1. Get rid of try-catch blocks that are unnecessary
2. Clearer separation of concerns - IllegalActionException thrown in Model, CommandException thrown in Commands

Also, IntervieweeList can be modified after finalising, only AttributeList and QuestionList cannot.

I made a few other comments as well!

File: src/main/java/seedu/address/logic/commands/EditIntervieweeCommand.java [hunk 3]
```diff
@@ -36,7 +36,7 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         IntervieweeList interviewees = model.getIntervieweeList();
         try {
-            interviewees.setName(identifier, updatedName);
+            interviewees.getInterviewee(identifier).setFullName(updatedName);
```
Comment (CornCobs) [inline src/main/java/seedu/address/logic/commands/EditIntervieweeCommand.java]:
```suggestion
            interviewees.setName(identifier, updatedName);
```

File: src/main/java/seedu/address/logic/commands/OpenReportCommand.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * OpenReportCommand describes the behavior when the
+ * client wants to open an interview report of an interviewee.
+ */
+public class OpenReportCommand extends Command {
+    public static final String COMMAND_WORD = "report";
+    public static final String MESSAGE_SUCCESS = "Interview report opened: %1$s";
+    public static final String MESSAGE_USAGE = "open " + COMMAND_WORD
+            + ": Opens the interview report of an interviewee."
+            + "Example: open " + COMMAND_WORD + " John Doe";
```
Comment (CornCobs) [inline src/main/java/seedu/address/logic/commands/OpenReportCommand.java]: Still says the command word is "report", but it should be "open"?

**Review by CornCobs — COMMENTED** [review]
A lot of repeated code in the validation for finalizing in all the commands, but I guess that can be refactored later. I would think some common method either in Command or some Util class would be ok.

File: src/main/java/seedu/address/logic/commands/OpenReportCommand.java (hunk 1 above)
Comment (agnesnatasya) [reply src/main/java/seedu/address/logic/commands/OpenReportCommand.java]: Okay done

**Pull request merged**